hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/client/HdfsClientConfigKeys.java

```diff
@@ -244,6 +244,10 @@
       "dfs.namenode.snapshot.capture.openfiles";
   boolean DFS_NAMENODE_SNAPSHOT_CAPTURE_OPENFILES_DEFAULT = false;
 
+  String DFS_NAMENODE_SNAPSHOT_TRASHROOT_ENABLED =
+      "dfs.namenode.snapshot.trashroot.enabled";
+  boolean DFS_NAMENODE_SNAPSHOT_TRASHROOT_ENABLED_DEFAULT = false;
+
   String DFS_PROVIDED_ALIASMAP_INMEMORY_RPC_ADDRESS =
```

Comment:
u may also want to add it in hdfs-default.xml or want to maintain it as an internal config? This ls leading to a test failure TestHdfsConfigFields.testCompareConfigurationClassAgainstXml.

I think its better to hide it and define the config in SnapshotManager itself similar to "dfs.namenode.snapshot.deletion.ordered"

Reply:
makes sense. will hide it from the client

Reply:
If we hide it, i would prefer to not add it in hdfs-default.xml as well. We can make this an internal config which is required only if u want to restrict renames out of snapshottable root which can be turned on only if a requirement like this arises in certain use cases.

Reply:
You are right. Done in 6b0a2585e837d7386fc52f58fc32e668d2440f5d.